homeassistant/components/daikin/__init__.py

```diff
@@ -4,112 +4,105 @@
-
-        if daikin_api_setup(hass, host) is None:
-            return
-
-        for component in COMPONENT_TYPES:
-            load_platform(hass, component, DOMAIN, discovery_info,
-                          config)
+    if DOMAIN not in config:
+        return True
+
+    hosts = config[DOMAIN].get(CONF_HOSTS)
+    if not hosts:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={'source': config.SOURCE_IMPORT}))
+    for host in hosts:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={'source': config.SOURCE_IMPORT},
+                data={
+                    KEY_HOST: host,
+                }))
+    return True
 
-    discovery.listen(hass, SERVICE_DAIKIN, discovery_dispatch)
 
-    for host in config.get(DOMAIN, {}).get(CONF_HOSTS, []):
-        if daikin_api_setup(hass, host) is None:
-            continue
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Establish connection with Daikin."""
+    conf = entry.data
+    daikin_api = await daikin_api_setup(hass, conf[KEY_HOST])
+    if not daikin_api:
+        return False
+    hass.data.setdefault(DOMAIN, {}).update({entry.entry_id: daikin_api})
+    await asyncio.wait([
+        hass.config_entries.async_forward_entry_setup(entry, component)
+        for component in COMPONENT_TYPES
+    ])
+    return True
 
-        discovery_info = {
-            'ip': host,
-            CONF_MONITORED_CONDITIONS:
-                config[DOMAIN][CONF_MONITORED_CONDITIONS]
-        }
-        load_platform(hass, 'sensor', DOMAIN, discovery_info, config)
 
+async def async_unload_entry(hass, config_entry):
+    """Unload a config entry."""
+    await asyncio.wait([
+        hass.config_entries.async_forward_entry_unload(config_entry, component)
+        for component in COMPONENT_TYPES
+    ])
+    hass.data[DOMAIN].pop(config_entry.entry_id)
+    if not hass.data[DOMAIN]:
+        hass.data.pop(DOMAIN)
     return True
 
 
-def daikin_api_setup(hass, host, name=None):
+async def daikin_api_setup(hass, host):
     """Create a Daikin instance only once."""
-    if DOMAIN not in hass.data:
-        hass.data[DOMAIN] = {}
-
-    api = hass.data[DOMAIN].get(host)
-    if api is None:
-        from pydaikin import appliance
-
-        try:
-            device = appliance.Appliance(host)
-        except timeout:
-            _LOGGER.error("Connection to Daikin could not be established")
-            return False
-
-        if name is None:
-            name = device.values['name']
-
-        api = DaikinApi(device, name)
+    from pydaikin.appliance import Appliance
+    try:
+        with async_timeout.timeout(10):
+            device = await hass.async_add_executor_job(Appliance, host)
+    except asyncio.TimeoutError:
+        _LOGGER.error("Connection to Daikin could not be established")
+        return False
```

Comment:
Return `None` instead. We would return False here if the positive flow would return True.